### PR TITLE
bootstrap CI

### PR DESCRIPTION
**Pull request by dericcrago**

### CODE REVIEW

File: changelogs/config.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  - Deprecated Features
+- - removed_features
+  - Removed Features (previously deprecated)
+- - security_fixes
+  - Security Fixes
+- - bugfixes
+  - Bugfixes
+- - known_issues
+  - Known Issues
+title: CHANGE THIS IN changelogs/config.yaml!
```
Comment (dmsimard) [inline changelogs/config.yaml:28]: title needs to be community.hashi_vault for generating the changelog with [antsibull-changelog](https://github.com/ansible-community/antsibull-changelog), like this: https://github.com/ansible-collections/community.kubevirt/pull/9

File: galaxy.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# See https://docs.ansible.com/ansible/latest/dev_guide/collections_galaxy_meta.html
+
+namespace: community
+name: hashi_vault
+version: 0.1.0
+readme: README.md
+authors:
+  - YOUR NAME (github.com/YOURGITHUB)
```
Comment (dmsimard) [inline galaxy.yml:8]: I guess we can use authors from the plugin ?
https://github.com/ansible-collections/community.hashi_vault/blob/8248977742411ddf22223dad253755377f96102f/plugins/lookup/hashi_vault.py#L12-L13

File: .github/workflows/ansible-test.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
+# README FIRST
+# 1. If you don't have unit tests remove that section
+# 2. If your collection depends on other collections ensure they are installed, see "Install collection dependencies"
+# If you need help please ask in #ansible-devel on Freenode IRC
```
Comment (dmsimard) [inline .github/workflows/ansible-test.yml:4]: @gundalow just checking, is #ansible-devel accurate ?

File: galaxy.yml (hunk 2 above)
Comment (dericcrago) [reply galaxy.yml]: agreed. thanks!

File: changelogs/config.yaml (hunk 1 above)
Comment (dericcrago) [reply changelogs/config.yaml]: updated. thanks!

**Comment by dericcrago** [comment]
CI is passing, but a couple of things don't feel right yet

- [ ] the requirement for `community.crypto` is called out in 3 different places
  - [ ] [.github/workflows/ansible-test.yml](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-f9f280a4139baf69c50e6fc8bd80487db74ffd4f1f817edc5972a3c288aa8486R100)
  - [ ] [tests/integration/targets/lookup_hashi_vault/runme.sh](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-ef8857ca47e164df036e31968579b1bc9474b29d9c911ed04925cbdc5b363179R5)
  - [ ] [tests/requirements.yml](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-3c15561a763d4c893c12e2ee08880b8fd1a011b7c38c81973835bdf522069695R3)
- [ ] integration tests are only running against [python 3.6](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-f9f280a4139baf69c50e6fc8bd80487db74ffd4f1f817edc5972a3c288aa8486R83)
- [ ] ignoring the 'line to long' sanity errors in the [examples](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-c47dc8233f99f31fc4a283e177d047b83d3df18d8e52f25a3ca3b375bffce0e9R264)
- [ ] additional integration targets (required by `tests/integration/targets/lookup_hashi_vault`)
  - [ ] `tests/integration/targets/setup_docker`
  - [ ] `tests/integration/targets/setup_epel`
  - [ ] `tests/integration/targets/setup_openssl`
  - [ ] `tests/integration/targets/setup_pkg_mgr`
  - [ ] `tests/integration/targets/setup_remote_constraints`
  - [ ] `tests/integration/targets/setup_remote_tmp_dir`

File: README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+# community.hashi_vault Collection
+<!-- Add CI and code coverage badges here. Samples included below. -->
+[![CI](https://github.com/ansible-collections/community.hashi_vault/workflows/CI/badge.svg?event=push)](https://github.com/ansible-collections/community.hashi_vault/actions) [![Codecov](https://img.shields.io/codecov/c/github/ansible-collections/community.hashi_vault)](https://codecov.io/gh/ansible-collections/community.hashi_vault)
+
+<!-- Describe the collection and why a user would want to use it. What does the collection do? -->
+
+## Tested with Ansible
+
+<!-- List the versions of Ansible the collection has been tested with. Must match what is in galaxy.yml. -->
```
Comment (dericcrago) [inline README.md:9]: the tested versions of ansible aren't listed in the `galaxy.xml` are they? Is this referencing something else?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+- [The Bullhorn (the Ansible Contributor newsletter)](https://us19.campaign-archive.com/home/?u=56d874e027110e35dea0e03c1&id=d6635f5420)
+- [Changes impacting Contributors](https://github.com/ansible-collections/overview/issues/45)
+
+## Licensing
+
+<!-- Include the appropriate license information here and a pointer to the full licensing details. If the collection contains modules migrated from the ansible/ansible repo, you must use the same license that existed in the ansible/ansible repo. See the GNU license example below. -->
+
+GNU General Public License v3.0 or later.
+
+See [LICENSE](https://www.gnu.org/licenses/gpl-3.0.txt) to see the full text.
```
Comment (felixfontein) [inline README.md:61]: https://github.com/ansible-collections/community.hashi_vault/blob/main/plugins/lookup/hashi_vault.py#L4 says the file is called `COPYING`. I guess you should adjust the plugin's header to mention `LICENSE`.

File: README.md (hunk 4 above)
Comment (felixfontein) [reply README.md:9]: It's not mentioned anywhere. (meta/runtime.yml should say the collection supports 2.9.10+ or something like that, I forgot what we had in the collection template.)

File: .github/workflows/ansible-test.yml (hunk 3 above)
Comment (gundalow) [reply .github/workflows/ansible-test.yml:4]: #ansible-devel is correct

**Comment by briantist** [comment]
> CI is passing, but a couple of things don't feel right yet
> 
> * [ ]  the requirement for `community.crypto` is called out in 3 different places
>   
>   * [ ]  [.github/workflows/ansible-test.yml](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-f9f280a4139baf69c50e6fc8bd80487db74ffd4f1f817edc5972a3c288aa8486R100)
>   * [ ]  [tests/integration/targets/lookup_hashi_vault/runme.sh](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-ef8857ca47e164df036e31968579b1bc9474b29d9c911ed04925cbdc5b363179R5)
>   * [ ]  [tests/requirements.yml](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-3c15561a763d4c893c12e2ee08880b8fd1a011b7c38c81973835bdf522069695R3)

Curious what Felix thinks, but I'm inclined to leave it for now and let it be something to improve upon over time?

> * [ ]  integration tests are only running against [python 3.6](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-f9f280a4139baf69c50e6fc8bd80487db74ffd4f1f817edc5972a3c288aa8486R83)

Is this because they didn't work with later python versions? Or it just wasn't being tested against them before? I would hope we would test against more versions.

> * [ ]  ignoring the 'line to long' sanity errors in the [examples](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-c47dc8233f99f31fc4a283e177d047b83d3df18d8e52f25a3ca3b375bffce0e9R264)

This is a tough one. I had a PR that started before the split to collections and wasn't merged until after, and I had to get real creative making existing examples (even ones I didn't touch) short enough to fit after changing references to FQ names. I brought this up in a docs meeting forever ago (as in earlier this year). I wish we had a longer limit on the doc strings, but ignoring might be going too far. 

I'm not looking forward to trimming more from examples as it limits what options can be shown used together, harming examples imo.

> * [ ]  additional integration targets (required by `tests/integration/targets/lookup_hashi_vault`)
>   
>   * [ ]  `tests/integration/targets/setup_docker`
>   * [ ]  `tests/integration/targets/setup_epel`
>   * [ ]  `tests/integration/targets/setup_openssl`
>   * [ ]  `tests/integration/targets/setup_pkg_mgr`
>   * [ ]  `tests/integration/targets/setup_remote_constraints`
>   * [ ]  `tests/integration/targets/setup_remote_tmp_dir`

I had no idea these were required, might take some time to untangle all that.

**Comment by felixfontein** [comment]
> > * [ ]   the requirement for `community.crypto` is called out in 3 different places
> >   
> >   * [ ]   [.github/workflows/ansible-test.yml](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-f9f280a4139baf69c50e6fc8bd80487db74ffd4f1f817edc5972a3c288aa8486R100)
> >   * [ ]   [tests/requirements.yml](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-3c15561a763d4c893c12e2ee08880b8fd1a011b7c38c81973835bdf522069695R3)

These are both ok and necessary. Eventually ansible-test should install the dependencies by itself (that's what the tests/requirements.yml is for), but right now it does not do that.

> >   * [ ]   [tests/integration/targets/lookup_hashi_vault/runme.sh](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-ef8857ca47e164df036e31968579b1bc9474b29d9c911ed04925cbdc5b363179R5)

It should never have been specified here, and it's not needed.

> > * [ ]   integration tests are only running against [python 3.6](https://github.com/ansible-collections/community.hashi_vault/pull/1/files#diff-f9f280a4139baf69c50e6fc8bd80487db74ffd4f1f817edc5972a3c288aa8486R83)

Why only 3.6? The template specifies the complete list: https://github.com/ansible-collections/collection_template/blob/main/.github/workflows/ansible-test.yml#L139-L148

> > * [ ]   additional integration targets (required by `tests/integration/targets/lookup_hashi_vault`)
> >   
> >   * [ ]   `tests/integration/targets/setup_docker`
> >   * [ ]   `tests/integration/targets/setup_epel`
> >   * [ ]   `tests/integration/targets/setup_openssl`
> >   * [ ]   `tests/integration/targets/setup_pkg_mgr`
> >   * [ ]   `tests/integration/targets/setup_remote_constraints`
> >   * [ ]   `tests/integration/targets/setup_remote_tmp_dir`
> 
> I had no idea these were required, might take some time to untangle all that.

setup_openssl is required here: https://github.com/ansible-collections/community.general/blob/main/tests/integration/targets/lookup_hashi_vault/playbooks/install_dependencies.yml#L4

This indirectly also requires setup_pkg_mgr, setup_remote_constraints and setup_remote_tmp_dir IIRC.

I have no idea why setup_docker and setup_epel show up in this list though (setup_epel is a dependency of setup_docker).

File: tests/integration/targets/lookup_hashi_vault/runme.sh [hunk 6]
```diff
@@ -2,6 +2,8 @@
 
 set -eux
 
+ansible-galaxy -vvv collection install community.crypto
+
```
Comment (felixfontein) [inline tests/integration/targets/lookup_hashi_vault/runme.sh:6]:
```suggestion
```

**Comment by briantist** [comment]
Making a dedicated comment about the python versions: 
Failures: https://github.com/ansible-collections/community.hashi_vault/runs/1433862247

Looks like it's openssl target that's failing:

```
[WARNING]: Updating cache and auto-installing missing dependency: python3-apt
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "allow_unauthenticated": false,
            "autoclean": false,
            "autoremove": false,
            "cache_valid_time": 0,
            "deb": null,
            "default_release": null,
            "dpkg_options": "force-confdef,force-confold",
            "force": false,
            "force_apt_get": false,
            "install_recommends": null,
            "name": "openssl",
            "only_upgrade": false,
            "package": [
                "openssl"
            ],
            "policy_rc_d": null,
            "purge": false,
            "state": "present",
            "update_cache": null,
            "update_cache_retries": 5,
            "update_cache_retry_max_delay": 12,
            "upgrade": null
        }
    },
    "msg": "Could not import python modules: apt, apt_pkg. Please install python3-apt package."
```

Interesting that at the top it says it's going to install the missing dependency, then it fails for not having it. Is there some reason that package would fail to install in this environment?

**Comment by felixfontein** [comment]
`setup_openssl` is for installing pyOpenSSL and cryptography on the OS docker images, not for the default docker images. I guess replacing the role by the `pip` module installing `cryptography` should be enough. (There's no real reason to install pyOpenSSL anyway.)

**Comment by briantist** [comment]
> `setup_openssl` is for installing pyOpenSSL and cryptography on the OS docker images, not for the `default` docker images. I guess replacing the role by the `pip` module installing `cryptography` should be enough. (There's no real reason to install pyOpenSSL anyway.)

Where is the appropriate place for adding the pip install? Is it a change I could do or help with in some way?

**Comment by gundalow** [comment]
> > `setup_openssl` is for installing pyOpenSSL and cryptography on the OS docker images, not for the `default` docker images. I guess replacing the role by the `pip` module installing `cryptography` should be enough. (There's no real reason to install pyOpenSSL anyway.)
> 
> Where is the appropriate place for adding the pip install? Is it a change I could do or help with in some way?

@briantist That would need adding into the integration tests

**Comment by briantist** [comment]
@felixfontein @dericcrago I'd be happy to mess around with the tests and try to get them to work, but I don't think I can push changes to this PR. Does it make sense to merge this, and then I can start a new one to try fixing up the openssl stuff and other python versions? 

Sorry if this doesn't make sense, new to working on splitting out a collection.

**Comment by dmsimard** [comment]
There are things to fix but this is a good first shot and CI is passing \o/

@dericcrago I'll merge this PR to unblock @briantist so he can iterate on the collection while you're out.

**Pull request merged**